### PR TITLE
solr-indexer

### PR DESCRIPTION
**Pull request by whereismyjetpack**
- Creates a `SolrSubmission` class that takes a `Submission` and makes it look like a solr document 
- adds helper methods to `Submission` to pull information / format information how solr needs it 
- Creates an index_submission method in the `SolrDataImportService`

I didn't touch any of the existing methods, and this code is not called yet from the application 

TODO 
The access fields aren't being populated, I think these fields are so that https://github.com/projectblacklight/blacklight-access_controls will work. I'm not sure if we are going to be using access_controls in new explore, we might want to write our own methods to simplify this? I'm not sure 

I'll need to write up some tests 

Wire these methods into the app 
  My thought for this is to put the new indexing behind a feature flag. something like 

```
   results = if ENV.fetch('SOLR_HOST', nil) 
                SolrDataImportService.new.index_submission(submission)
              else
                SolrDataImportService.new.delta_import
              end
```
where we are using the delta_import etc.

Once we migrate to solrcloud everywhere, we can remove all the dead code. 

Extra Credit might be to wrap the SolrDataImport into a background job so we can do parallellism on full imports / not hang the clients. 

Sorry for the wall of text. I wanted to put this out early for review.

### CODE REVIEW

File: app/models/submission.rb [hunk 1]
```diff
@@ -251,6 +251,83 @@ def committee_email_list
     list.uniq
   end
```
Comment (ajkiessl) [inline app/models/submission.rb:253]: I'm wondering if we can refactor these methods into a `SubmissionSchema` class kinda like we do in ScholarSphere.  Then implement the `SubmissionSchema` object in `SolrSubmission`.  `Submission` is getting a bit large and unwieldy.  Putting these methods elsewhere I think would make things cleaner.  I can take a stab at this if you're busy.

**Review by ajkiessl — CHANGES_REQUESTED** [review]
Overall I think this looks great!  I added two comments for potential changes.

**Comment by ajkiessl** [comment]
Oh and, as you mentioned above, we'll need tests for most of this.

**Comment by whereismyjetpack** [comment]
I really like the SimpleDelegator, i didn't know you could do that. 🎉

**Review by ajkiessl — APPROVED** [review]
This looks good to me, if you're ready to merge.

**Comment by whereismyjetpack** [comment]
LGTM 🎉

**Pull request merged**